spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/api/AfterAdvisor.java

```diff
@@ -77,21 +72,18 @@ public enum StreamResponseMode {
 
 	}
 
-	default StreamResponseMode getStreamResponseMode() {
-		return StreamResponseMode.ON_FINISH_ELEMENT;
+	default AfterStreamMode getAfterStreamMode() {
+		return AfterStreamMode.ON_FINISH_ELEMENT;
```

Comment:
Would it not be more natural if a default behaviour was to apply the advisor per each chunk in the stream using the imperative implementation? Or perhaps applying it over the aggregation? If there's no easy answer, perhaps picking a default should not happen at all and stream advisors should not belong to the imperative type at all?